### PR TITLE
Series Details Modal

### PR DESCRIPTION
**Pull request by isabellakutger**
This PR implements the modal for displaying and editing series details.
The modal provides the following pages:
- metadata (edit possible)
- access permissions (edit possible)
- theme (edit possible)
- feeds (no edit possible)

### CODE REVIEW

File: modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js [hunk 1]
```diff
@@ -259,7 +259,7 @@ const ResourceDetailsAccessPolicyTab = ({ resourceId, header, t, policies, fetch
                                                                             {(aclTemplates && aclTemplates.length > 0) ? (
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js:225]: Remove `createPolicy`, line 45 to 53 and import the one from the utilities

File: modules/admin-ui-picard/app/src/components/events/partials/modals/SeriesDetailsModal.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <>
+            <div className="modal-animation modal-overlay"/>
+            <section className="modal modal-animation" id="series-details-modal">
+                <header>
+                    <a className="fa fa-times close-modal" onClick={() => close()}/>
+                    <h2>{t('EVENTS.SERIES.DETAILS.HEADER', { resourceId: seriesTitle })}</h2>
+                </header>
+
+                <SeriesDetails seriesId={seriesId}
+                               close={close}/>
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/events/partials/modals/SeriesDetailsModal.js]: The `close` doesn't seem to be used in `SeriesDetails`

File: modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsAccessTab.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import {useTranslation} from "react-i18next";
+import {connect} from "react-redux";
+import ResourceDetailsAccessPolicyTab from "../../../shared/modals/ResourceDetailsAccessPolicyTab";
+import {getSeriesDetailsAcl} from "../../../../selectors/seriesDetailsSelectors";
+import {fetchSeriesDetailsAcls, updateSeriesAccess} from "../../../../thunks/seriesDetailsThunks";
+
+/**
+ * This component manages the access policy tab of the series details modal
+ */
+const SeriesDetailsAccessTab = ({ seriesId, header, policies, fetchAccessPolicies, fetchHasActiveTransactions,
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsAccessTab.js]: `fetchHasActiveTransactions` is not used (remove if not needed)

File: modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsAccessTab.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    const { t } = useTranslation();
+
+    return (
+        <ResourceDetailsAccessPolicyTab resourceId={seriesId}
+                                        header={header}
+                                        t={t}
+                                        buttonText={'EVENTS.SERIES.DETAILS.ACCESS.ACCESS_POLICY.LABEL'}
+                                        policies={policies}
+                                        fetchAccessPolicies={fetchAccessPolicies}
+                                        saveNewAccessPolicies={saveNewAccessPolicies} />
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsAccessTab.js:22]: Add `descriptionText={t("EVENTS.SERIES.NEW.ACCESS.ACCESS_POLICY.DESCRIPTION")}` (`descriptionText` was added during refactoring of `ResourceDetailsAccessPolicyTab`)

File: modules/admin-ui-picard/app/src/components/events/partials/modals/SeriesDetails.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            </nav>
+
+            {/* render modal content depending on current page */}
+            <div>
+                {page === 0 && (
+                    <DetailsMetadataTab metadataFields={metadataFields}
+                                        resourceId={seriesId}
+                                        header={tabs[page].tabNameTranslation}
+                                        buttonLabel={updateSeries}
+                                        updateResource='EVENTS.SERIES.DETAILS.METADATA.REPLACE_SERIES_METADATA'/>
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/events/partials/modals/SeriesDetails.js]: I think `buttonLabel` and `updateResource` were switched here. shouldn't it be `buttonLabel='EVENTS.SERIES.DETAILS.METADATA.REPLACE_SERIES_METADATA'` and `updateResource={updateSeries}`?

```suggestion
                                        buttonLabel='EVENTS.SERIES.DETAILS.METADATA.REPLACE_SERIES_METADATA'
                                        updateResource={updateSeries}/>
```

File: modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                ...state,
+                isLoading: false,
+                feeds: seriesFeeds
+            }
+        }
+        case LOAD_SERIES_DETAILS_THEME_SUCCESS: {
+            const { seriesTheme } = payload;
+            return {
+                ...state,
+                isLoading: false,
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js:69]: This is used in the thunk without a `LOAD_SERIES_DETAILS_IN_PROGRESS` or `LOAD_SERIES_DETAILS_FAILURE`. I think in that case the `isLoading: false` shouldn't be here

File: modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    let data = new URLSearchParams();
+    data.append("themeId", themeId);
+
+    axios.put(`admin-ng/series/${id}/theme`, data)
+        .then(response => {
+            let themeResponse = response.data;
+
+            let seriesTheme = (transformToIdValueArray(themeResponse))[0].value;
+
+            dispatch(loadSeriesDetailsThemeSuccess(seriesTheme));
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js]: See also comment in modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js
Either add `dispatch(loadSeriesDetailsInProgress())` and `dispatch(loadSeriesDetailsFailure())` here, or don't change `isLoading` for this action in the reducer

File: modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+
+export const updateSeriesAccess = (id, policies) => async (dispatch) => {
+    let data = new URLSearchParams();
+
+    let acls = {
+        acl: {
+            ace: policies
+        }
+    }
+    data.append("acl", JSON.stringify(acls));
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js]: After the refactoring of `ResourceDetailsAccessPolicyTab` line 206 to 210 will no longer be needed. this line can be `    data.append("acl", JSON.stringify(policies));`

File: modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+
+        await axios.put(`/admin-ng/series/${id}/metadata`, data);
+
+        // updated metadata in series details redux store
+        let seriesMetadata = {
+            flavor: metadataInfos.flavor,
+            title: metadataInfos.title,
+            fields: fields
+        };
+        dispatch(loadSeriesDetailsMetadataSuccess(seriesMetadata));
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js]: See also comment in modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js
Either add `dispatch(loadSeriesDetailsInProgress())` and `dispatch(loadSeriesDetailsFailure())` here, or don't change `isLoading` for this action in the reducer

File: modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+            return {
+                ...state,
+                isLoading: true
+            }
+        }
+        case LOAD_SERIES_DETAILS_METADATA_SUCCESS: {
+            const { seriesMetadata } = payload;
+            return {
+                ...state,
+                isLoading: false,
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js:39]: This is used in the thunk without a `LOAD_SERIES_DETAILS_IN_PROGRESS` or `LOAD_SERIES_DETAILS_FAILURE`. I think in that case the `isLoading: false` shouldn't be here

**Review by SimoneSchmidt — CHANGES_REQUESTED** [review]

File: modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+    data.append("override", true);
+
+    return axios.post(`admin-ng/series/${id}/access`, data, {
+        headers: {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+    })
+        .then(res => {
+            console.log(res);
+            dispatch(addNotification('info', 'SAVED_ACL_RULES'));
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js]: This Notification should be displayed inside the modal (on the top of the tab), not in the lower right corner

File: modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+            return true;
+        })
+        .catch(res => {
+            console.log(res);
+            dispatch(addNotification('error', 'ACL_NOT_SAVED'));
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/thunks/seriesDetailsThunks.js]: This Notification should be displayed inside the modal (on the top of the tab), not in the lower right corner

File: modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsThemeTab.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+                                                <button type="submit"
+                                                        onClick={() => formik.handleSubmit()}
+                                                        disabled={!checkValidity(formik)}
+                                                        className={cn("submit",
+                                                            {
+                                                                active: checkValidity(formik),
+                                                                inactive: !checkValidity(formik)
+                                                            }
+                                                        )}>
+                                                    {t('EVENTS.SERIES.DETAILS.METADATA.REPLACE_SERIES_THEME')}
```
Comment (SimoneSchmidt) [inline modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/SeriesDetailsThemeTab.js:74]: I'm not shure, if this button works as it should. When trying it out, only the first time of changing the theme would work. When choosing a different theme afterwards and pressing the button again, nothing would happen.

**Review by SimoneSchmidt — CHANGES_REQUESTED** [review]

File: modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js (hunk 6 above)
Comment (isabellakutger) [reply modules/admin-ui-picard/app/src/reducers/seriesDetailsReducers.js:69]: see below

**Review by SimoneSchmidt — APPROVED** [review]

**Pull request merged**